__tests__/webhooks.ts

```diff
@@ -391,3 +391,49 @@ describe('POST /webhooks/customerio/reporting', () => {
     expect(body.success).toEqual(true);
   });
 });
+
+describe('POST /webhooks/customerio/promote', () => {
+  const timestamp = Math.floor(Date.now() / 1000);
+  const payload = {
+    userId: 'abc',
+    postId: 'def',
+  };
+
+  const hmac = createHmac(
+    'sha256',
+    process.env.CIO_REPORTING_WEBHOOK_SECRET as string,
+  );
+  hmac.update(`v0:${timestamp}:${JSON.stringify(payload)}`);
+  const hash = hmac.digest().toString('hex');
+
+  it('should return 403 when no x-cio-timestamp header', async () => {
+    const { body } = await request(app.server)
+      .post('/webhooks/customerio/promote')
+      .set('x-cio-signature', '123')
+      .send(payload)
+      .expect(403);
+
+    expect(body.error).toEqual('Invalid signature');
+  });
+
+  it('should return 403 when no x-cio-signature header', async () => {
+    const { body } = await request(app.server)
+      .post('/webhooks/customerio/promote')
+      .set('x-cio-timestamp', '123')
+      .send(payload)
+      .expect(403);
+
+    expect(body.error).toEqual('Invalid signature');
+  });
+
+  it('should return 200 when signature is valid', async () => {
+    const { body } = await request(app.server)
+      .post('/webhooks/customerio/promote')
+      .set('x-cio-timestamp', timestamp.toString())
+      .set('x-cio-signature', hash)
+      .send(payload)
+      .expect(200);
```

Comment:
Tried to write the test, but not sure how to check if message was posted to pubsub.

Reply:
Take a look at these.
We don't check that it actually get sent on pubsub, but check that the function that sends it, gets called.

https://github.com/dailydotdev/daily-api/blob/507e28917cf6a0e5801dacda75c6732e9c09cea5/__tests__/workers/cdc/primary.ts#L158

https://github.com/dailydotdev/daily-api/blob/507e28917cf6a0e5801dacda75c6732e9c09cea5/__tests__/workers/cdc/primary.ts#L788-L809